api/v1/handlers.go

```diff
@@ -613,6 +613,35 @@ func (h *Handler) ListValidators(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (h *Handler) RuntimeListBlocks(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	blocks, err := h.client.Blocks(ctx, r)
+	if err != nil {
+		h.logAndReply(ctx, "failed to list blocks", w, err)
+		h.metrics.RequestCounter(r.URL.Path, "failure", "database_error").Inc()
+		return
+	}
+
+	resp, err := json.Marshal(blocks)
+	if err != nil {
+		h.logAndReply(ctx, "failed to marshal blocks", w, err)
+		h.metrics.RequestCounter(r.URL.Path, "failure", "serde_error").Inc()
+		return
+	}
+
+	w.Header().Set("content-type", "application/json")
+	if _, err := w.Write(resp); err != nil {
+		h.logger.Error("failed to write response",
+			"request_id", ctx.Value(storage.RequestIDContextKey),
+			"error", err,
+		)
+		h.metrics.RequestCounter(r.URL.Path, "failure", "http_error").Inc()
+	} else {
+		h.metrics.RequestCounter(r.URL.Path, "success").Inc()
+	}
```

Comment:
Can this be abstracted away? So we call it from here with something like
```go
h.writeJsonResponse(w, blocks, r.URL.Path, ctx)
```

You can backport it to non-emerald handlers if you're feeling the least bit OCD, but even just using it for Emerald handlers would set a good precedent.

Reply:
yeah, I'm leaning towards that too. nexus had something comparable

Reply:
filing #196